src/java.base/share/classes/java/lang/Class.java

```diff
@@ -790,8 +791,9 @@ public T newInstance()
      * @return  {@code true} if this {@code Class} object represents an interface;
      *          {@code false} otherwise.
      */
-    @IntrinsicCandidate
-    public native boolean isInterface();
+    public boolean isInterface() {
+        return Modifier.isInterface(getModifiers());
```

Comment:
Is there any reason to go through the getter?
```suggestion
        return Modifier.isInterface(modifiers);
```

Reply:
You are right, using the field directly is indeed better.

Reply:
I don't use the field directly because the field is a short and getModifiers makes it into Modifiers.

Reply:
Indeed, even though this checks for the specific bit so widening has no effect, it is better to be cautious here.

Reply:
> I don't use the field directly because the field is a short and getModifiers makes it into Modifier.

But getModifiers() returns `int`, not `Modifier` (which is all static).